backend/conversion_adapter.go

```diff
@@ -2,27 +2,82 @@ package backend
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 
 	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
 )
 
 type conversionSDKAdapter struct {
-	handler ConversionHandler
+	handler                ConversionHandler
+	queryConversionHandler QueryConversionHandler
```

Comment:
Nit: Would be good to document/note that this is optional (just for the sake of knowing not to skip nil checks if anyone is going to change any code here)